File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/commands/PayoutResponseReceivedCommand.java
package com.hedvig.paymentservice.domain.trustlyOrder.commands;

import java.util.UUID;
import javax.money.MonetaryAmount;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

import lombok.Value;

@Value
public class PayoutResponseReceivedCommand {
    @TargetAggregateIdentifier
    UUID hedvigOrderId;

    String trustlyOrderId;
    MonetaryAmount amount;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/commands/PayoutResponseReceivedCommand.kt
package com.hedvig.paymentservice.domain.trustlyOrder.commands

import org.axonframework.commandhandling.TargetAggregateIdentifier
import java.util.*
import javax.money.MonetaryAmount

data class PayoutResponseReceivedCommand(
  @TargetAggregateIdentifier
  val hedvigOrderId: UUID,
  val trustlyOrderId: String,
  val amount: MonetaryAmount
)
